### PR TITLE
Replace UUID.randomUUID() with a faster implementation

### PR DESCRIPTION
**Pull request by tmancill**
Instantiating a TelemetryBatch calls `UUID.randomUUID()`, which in turn uses `SecureRandom` to generate 16 random bytes.  When generating TelemetryBatch at high rates, this can exhaust the entropy source and is computationally expensive.

This PR replaces all calls to `UUID.randomUUID()` with a utility method that generates UUIDs using `ThreadLocalRandom.current().nextLong()`.

### CODE REVIEW

File: telemetry-core/src/main/java/com/newrelic/telemetry/util/Utils.java [hunk 1]
```diff
@@ -25,4 +28,8 @@ public static String verifyNonBlank(String input, String message)
   public static <E> E verifyNonNull(E input) throws IllegalArgumentException {
     return verifyNonNull(input, "input cannot be null");
   }
+
+  public static UUID generateUUID() {
+    return new UUID(ThreadLocalRandom.current().nextLong(), ThreadLocalRandom.current().nextLong());
```
[the remark this answers is not in the document]
Comment (tbradellis) [reply telemetry-core/src/main/java/com/newrelic/telemetry/util/Utils.java:33]: After chat we determined that UUID.randomUUID() is used out of convenience, not out of any guarantees of cryptographically security.
However we may want to later  provide a separate constructor that supports passing in a UUID, or a UUID generator to give the option.

**Review by tbradellis — APPROVED** [review]
lgtm

**Review by meiao — APPROVED** [review]

**Comment by Steven-M-Brown** [comment]
Going to necro this PR.

While I understand (and actually agree with) the desire to update this the way this is done has a subtle bug in that you are generating a V4 random UUID but ignoring the reserved bits in the standard specifying the version/variant. In most cases this won't be an issue but if some system tries to do something with that UUID it may cause issues.

Should be an easy fix, just clear/set the appropriate bits after generating the values.

**Comment by tmancill** [comment]
Hi @Steven-M-Brown . Thanks for reviewing and bringing this up.  It's a little strange that the `UUID()` constructor doesn't handle adherence - that is that `UUID.randomUUID()` is the only code that generates a V4 random UUID, but that's an issue for the JDK.  I will put up a PR to implement the version logic from `randomUUID()`.

Just curious - did you run into this because a downstream system is expecting V4 UUIDs?

For reference:  https://www.ietf.org/rfc/rfc4122.txt

**Comment by Steven-M-Brown** [comment]
I would imagine the JDK doesn't because they don't want to cover every case and that constructor could conceivable be used to implement any unsupported variant of it whereas `randomUUID()` does because that is pretty much the current definition of a type 4 UUID as far as I know.

I just happened to notice because I was looking through the change logs before validating the new agent version in our infrastructure.

File: telemetry-core/src/main/java/com/newrelic/telemetry/util/Utils.java (hunk 1 above)
[the remark this answers is not in the document]
Comment (tmancill) [reply telemetry-core/src/main/java/com/newrelic/telemetry/util/Utils.java:33]: This concern is addressed in #300.

**Pull request merged**